R/zchunk_L2111.ag_nonco2_IRR.R

```diff
@@ -0,0 +1,171 @@
+      select(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology, year,
+             Non.CO2, input.emissions)
+
+    # L2111.AGREmissions: ag AGR emissions in all regions
+    L2111.AGREmissions <- L1221.ghg_tg_R_agr_C_Y_GLU_IRR %>%
+      filter(year %in% emissions.MODEL_BASE_YEARS) %>%
+      # Add region names and Ag columns
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      rename(AgSupplySector = GCAM_commodity,
+             input.emissions = value) %>%
+      mutate(AgSupplySubsector = paste(AgSupplySector, GLU, sep = "_"),
+             AgProductionTechnology = paste(AgSupplySubsector, Irr_Rfd, sep = "_"),
+             input.emissions = round(input.emissions, emissions.DIGITS_EMISSIONS)) %>%
+      select(region, AgSupplySector, AgSupplySubsector, AgProductionTechnology, year,
+             Non.CO2, input.emissions)
+
+    # L2111.AGRBio, L2111.AWB_BCOC_EmissCoeff, L2111.nonghg_max_reduction, L2111.nonghg_steepness: repeat by irr/rfd and copy
+    Irr_Rfd <- tibble(Irr_Rfd = c("IRR", "RFD"))
+    L2111.AGRBio <- repeat_add_columns(L211.AGRBio, Irr_Rfd) %>%
+      mutate(AgProductionTechnology = paste(AgSupplySubsector, Irr_Rfd, sep = "_")) %>%
+      select(-Irr_Rfd)
+
+    L2111.AWB_BCOC_EmissCoeff <- repeat_add_columns(L211.AWB_BCOC_EmissCoeff, Irr_Rfd) %>%
+      mutate(AgProductionTechnology = paste(AgSupplySubsector, Irr_Rfd, sep = "_")) %>%
+      select(-Irr_Rfd)
+
+    L2111.nonghg_max_reduction <- repeat_add_columns(L211.nonghg_max_reduction, Irr_Rfd) %>%
+      mutate(AgProductionTechnology = paste(AgSupplySubsector, Irr_Rfd, sep = "_")) %>%
+      select(-Irr_Rfd)
+
+    L2111.nonghg_steepness <- repeat_add_columns(L211.nonghg_steepness, Irr_Rfd) %>%
+      mutate(AgProductionTechnology = paste(AgSupplySubsector, Irr_Rfd, sep = "_")) %>%
+      select(-Irr_Rfd)
+      # ===================================================
+
+      # Produce outputs
+      L2111.AWBEmissions %>%
+        add_title("Agricultural Waste Burning Emissions") %>%
+        add_units("Tg") %>%
+        add_comments("Add region and SO2 name to AWB emissions") %>%
+        add_legacy_name("L2111.AWBEmissions") %>%
+        add_precursors("L1211.nonco2_tg_R_awb_C_Y_GLU_IRR", "common/GCAM_region_names", "emissions/A_regions") ->
+        L2111.AWBEmissions
+      L2111.AGREmissions %>%
+        add_title("Agricultural Emissions") %>%
+        add_units("Tg") %>%
+        add_comments("Added region to AGR emissions") %>%
+        add_legacy_name("L2111.AGREmissions") %>%
+        add_precursors("temp-data-inject/L1221.ghg_tg_R_agr_C_Y_GLU_IRR", "water/basin_to_country_mapping",
+                       "common/GCAM_region_names") ->
+        L2111.AGREmissions
+      L2111.AGRBio %>%
+        add_title("Bio N2O Coefficients by region and technology") %>%
+        add_units("Unitless") %>% # should this be labeled in emissions/A_regions
```

Comment:
The data originated from A_regions, but the units aren't labeled there, so I wasn't sure what to put. I opened an issue (#559). I'm not sure I actually meant to leave that comment in though.